clients/fides-js/src/lib/consent-constants.ts

```diff
@@ -6,11 +6,18 @@ import {
 } from "./consent-types";
 import { LOCALE_REGEX } from "./i18n/i18n-constants";
 
-// Regex to validate an ISO-3166 location string, which must:
-// 1) Start with a 2-3 letter country code (e.g. "US", "USA", "EEA")
-// 2) Optionally end with a 1-3 alphanumeric character region code (e.g. "CA", "123", "X")
-// 3) Separated by a dash (e.g. "US-CA")
-export const VALID_ISO_3166_LOCATION_REGEX = /^[a-z]{2,3}(-[a-z0-9]{1,3})?$/i;
+/**
+ * Regex to validate a [ISO 3166-2](https://en.wikipedia.org/wiki/ISO_3166-2) code:
+ * 1. Starts with a 2 letter country code (e.g. "US", "GB") (see [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2))
+ * 2. (Optional) Ends with a 1-3 alphanumeric character region code (e.g. "CA", "123", "X") (see [ISO 3166-2](https://en.wikipedia.org/wiki/ISO_3166-2))
+ * 3. Country & region codes must be separated by a hyphen (e.g. "US-CA")
+ *
+ * Fides also supports a special `EEA` geolocation code to denote the European
+ * Economic Area; this is not part of ISO 3166-2, but is supported for
+ * convenience.
+ */
+export const VALID_ISO_3166_LOCATION_REGEX =
+  /^(?:([a-z]{2})(-[a-z0-9]{1,3})?|(eea))$/i;
```

Comment:
This regex is getting a bit more complicated but really only because of the `()` characters that add some visual noise. You can try it here to see more of an explanation: https://regex101.com/r/lYuAGj/2